lib/runtime/procs/join.js

```diff
@@ -801,11 +802,20 @@ class join extends base {
         return joinpts;
     }
     emit_result(points) {
-        // emit the points, possibly bracketed by marks, and update emitter state.
+        // Emit the points, possibly bracketed by marks, and update emitter state.
+        // Mark interval will be the distance between group start and complete times.
         var leader = this.batched_leader() ;
         var join_time = this.outer ? this.ins[this.outer - 1].live_time() : this.max_live_time();
+        var interval;
+
         if (leader) {
-            this.maybe_emit_mark(leader.live().start_time);
+            var live_leader = leader.live();
+            interval = live_leader.interval;
+            interval = interval || live_leader.complete_time.subtract(live_leader.start_time);
```

Comment:
When the mark in a live batched input does not contain an interval. This could happen for the first mark (group). But it should be possible to avoid this case by taking the interval from the 'end-of-batch' mark of the first batch, instead of the initial one.